assembl/static2/js/app/components/search.jsx

```diff
@@ -199,181 +211,231 @@ const highlightedLSOrTruncatedLS = (hit, field, locale) => {
   return '';
 };
 
-const DumbPostHit = (props) => {
-  const locale = props.locale;
-  const source = props.result._source;
-  const subject = highlightedLSOrTruncatedLS(props.result, 'subject', locale);
-  const body = highlightedLSOrTruncatedLS(props.result, 'body', locale);
-  return (
-    <div className={props.bemBlocks.item().mix(props.bemBlocks.container('item'))}>
-      <ImageType type={props.result._type} className={props.bemBlocks.item('imgtype')} />
-      <div className={props.bemBlocks.item('title')}>
-        <Link to={getUrl(props.result)} dangerouslySetInnerHTML={{ __html: subject }} />
-      </div>
-      <div className={props.bemBlocks.item('content')}>
-        <p dangerouslySetInnerHTML={{ __html: body }} />
-        <div>
-          <div title={I18n.t('search.like')} className="emoticon LikeSentimentOfPost" />
-          <div className="emoticonValue">{source.sentiment_counts.like}</div>
-          <div title={I18n.t('search.disagree')} className="emoticon DisagreeSentimentOfPost" />
-          <div className="emoticonValue">{source.sentiment_counts.disagree}</div>
-          <div title={I18n.t('search.dont_understand')} className="emoticon DontUnderstandSentimentOfPost" />
-          <div className="emoticonValue">{source.sentiment_counts.dont_understand}</div>
-          <div title={I18n.t('search.more_info')} className="emoticon MoreInfoSentimentOfPost" />
-          <div className="emoticonValue">{source.sentiment_counts.more_info}</div>
-        </div>
-      </div>
-      <PublishedInfo
-        className={props.bemBlocks.item('info')}
-        date={source.creation_date}
-        userId={source.creator_id}
-        userName={source.creator_name}
-      />
+const BaseHit = ({ bemBlocks, imageType, onLinkClick, title, url, renderBody, renderFooter }) => (
+  <div className={bemBlocks.item().mix(bemBlocks.container('item'))}>
+    <ImageType type={imageType} className={bemBlocks.item('imgtype')} />
+    <div className={bemBlocks.item('title')}>
+      {url ? <Link onClick={onLinkClick} to={url} dangerouslySetInnerHTML={{ __html: title }} /> : <p dangerouslySetInnerHTML={{ __html: title }} />}
     </div>
+    {renderBody && <div className={bemBlocks.item('content')}>{renderBody()}</div>}
+    <div className={bemBlocks.item('info')}>{renderFooter()}</div>
+  </div>
+);
+
+const PostHit = ({ bemBlocks, collapseSearch, locale, result }) => {
+  const source = result._source;
+  const subject = highlightedLSOrTruncatedLS(result, 'subject', locale);
+  const body = highlightedLSOrTruncatedLS(result, 'body', locale);
+  const published = {};
+  return (
+    <BaseHit
+      bemBlocks={bemBlocks}
+      imageType={result._type}
+      title={subject}
+      url={getUrl(result)}
+      onLinkClick={collapseSearch}
+      published={published}
+      renderBody={() => (
+        <React.Fragment>
```

Comment:
What's the point of putting Fragment here when this component is not reusable by other components?

Reply:
Or is this just a way to avoid wrapping the children in another DOM element?

Reply:
yes, I don't want an additional unecessary `<div>`. With `<React.Fragment>`, my snapshots stayed unchanged after refactoring which is comforting ;-)